frontend/src/components/CreateRecipeForm/CreateRecipeForm.js

```diff
@@ -42,7 +44,32 @@ const EditRecipeForm = ({ history }) => {
   const [selectedTagsWithId, setSelectedTagsWithId] = useState([]);
   const [refreshTags, setRefreshTags] = useState(true);
   const [ingredients, setIngredients] = useState(['']);
+  const [isEditing, setIsEditing] = useState(false);
```

Comment:
What is the purpose of `isEditing`? If we're trying to prevent the `useEffect`s from overwriting changes, we can make those hooks [fire conditionally](https://reactjs.org/docs/hooks-reference.html#conditionally-firing-an-effect).

Reply:
@vhoof The way the ingredients and directions are set up, without isEditing, the respective handleChange functions won't allow the state to change/won't allow any editing in the text inputs.

Reply:
Yeah, I think that's because the useEffect hooks fire every time the component gets updated (which it will every time you change `directions`). If you wrote those like:

```
useEffect(() => {
  // do some stuff
}, [currentRecipe]);
```

This would cause that hook to only fire when the `currentRecipe` object changes, which it shouldn't unless you change the URL. Unless there's something else going on I'm not understanding?

Reply:
@nickharriscodes Did you look into this suggestion? I think conditional firing of the hooks would be the preferred way forward here.

Reply:
@H12 ran into some errors... trying again right now